lib/loader.js

```diff
@@ -41,8 +41,8 @@ module.exports = function (content) {
   this.cacheable()
   var isServer = this.options.target === 'node'
   var loaderContext = this
-  var options = this.vue || this.options.vue || {}
   var query = loaderUtils.parseQuery(this.query)
+  var options = this.options.vue || query || {}
```

Comment:
@yyx990803 Alright, so the issue is that query is an empty object by default, and therefore needs to be at the end of the list.
Perhaps we should try merging `this.options.vue` and `this.vue` into query?